manim/renderer/opengl_renderer.py

```diff
@@ -244,12 +243,26 @@ def init_scene(self, scene):
         self.scene = scene
         if not hasattr(self, "window"):
             if config["preview"]:
+                from .opengl_renderer_window import Window
+
                 self.window = Window(self)
                 self.context = self.window.ctx
                 self.frame_buffer_object = self.context.detect_framebuffer()
             else:
+                context_settings = {}
+
+                try:
+                    from IPython import get_ipython
+
+                    if "google.colab" in str(get_ipython()):
```

Comment:
Changing the backend to egl in more cases would (I think) also allow OpenGL rendering to be used in other remote settings (e.g., https://try.manim.community).

```suggestion
                    if get_ipython() is not None:
```

Reply:
Not so sure about this, if I understand correctly EGL only works on Linux, so won't this break Windows and Mac notebooks?

Reply:
I don't think the EGL Backend is available on all IPython shells. At least it's not on my local Windows machine.

Reply:
Just making Binder work will still leave other remote settings broken I fear. I'd prefer if we checked for egl support and if it is not available, fell back to the current default. (I don't have a particularly strong opinion about this though, and would be fine with "just" fixing binder.)

Reply:
> 
> 
> Just making Binder work will still leave other remote settings broken I fear. I'd prefer if we checked for egl support and if it is not available, fell back to the current default. (I don't have a particularly strong opinion about this though, and would be fine with "just" fixing binder.)

Is there any reason you'd "prefer" EGL over the current default? Since there may be significant speed differences, I'd prefer if someone with more OpenGL knowledge weighed in on which is "better" as a default. In the meantime I think we could "just" fix binder with this PR, or even perhaps leave the binder fixing to a different one.

Reply:
As far as I understand, the current default does not work in a headless environment / without a virtual frame buffer. I did a substantial amount of investigation into this while looking into preparing OpenGL rendering for RTD. EGL, as per Naveen's comment on Discord, allows to work without the virtual frame buffer -- which is why I think it is the more suitable (default!) choice when working with Jupyter.

I prefer a working, but potentially slower default backend setting, than one that doesn't work at all.

Reply:
> 
> 
> As far as I understand, the current default does not work in a headless environment / without a virtual frame buffer. I did a substantial amount of investigation into this while looking into preparing OpenGL rendering for RTD. EGL, as per Naveen's comment on Discord, allows to work without the virtual frame buffer -- which is why I think it is the more suitable (default!) choice when working with Jupyter (because I usually think of Jupyter in such a headless, running-on-some-remote-server setting).
> 
> I prefer a working, but potentially slower default backend setting, over one that doesn't work at all.
[Moderngl choose a backend based on the OS you're running](https://github.com/moderngl/glcontext/blob/b4aa30c3f9cab27ea606344d5d4d867e7bc8b974/glcontext/__init__.py#L22-L41). Why override this setting? We could let moderngl try to choose a backend first, and if it fails, then choose EGL ourselves.

Reply:
That sounds good to me. Then the only question left to answer is how to check whether a given backend is available.

Reply:
Great, that's in line with your suggestion. Practically, this means that the context creation should probably be moved into a `try` block which, should it fail, switches the backend to EGL. (And that should then also fix the situation for Binder and Colab, I hope.)